crates/host/src/wasmbus/mod.rs

```diff
@@ -1094,89 +980,13 @@ impl Host {
-                    .revision(
-                        component
-                            .claims()
-                            .and_then(|claims| claims.metadata.as_ref())
-                            .and_then(|jwt::Component { rev, .. }| *rev)
-                            .unwrap_or_default(),
-                    );
-                // Add name if present
-                if let Some(name) = component
-                    .claims()
-                    .and_then(|claims| claims.metadata.as_ref())
-                    .and_then(|metadata| metadata.name.as_ref())
-                    .cloned()
-                {
-                    description = description.name(name);
-                };
-
-                Some(
-                    description
-                        .build()
-                        .expect("failed to build component description: {e}"),
-                )
-            })
-            .collect()
-            .await;
-
-        let providers: Vec<_> = self
-            .providers
-            .read()
-            .await
-            .iter()
-            .map(
-                |(
-                    provider_id,
-                    Provider {
-                        annotations,
-                        claims_token,
-                        image_ref,
-                        ..
-                    },
-                )| {
-                    let mut provider_description = ProviderDescription::builder()
-                        .id(provider_id)
-                        .image_ref(image_ref);
-                    if let Some(name) = claims_token
-                        .as_ref()
-                        .and_then(|claims| claims.claims.metadata.as_ref())
-                        .and_then(|metadata| metadata.name.as_ref())
-                    {
-                        provider_description = provider_description.name(name);
-                    }
-                    provider_description
-                        .annotations(
-                            annotations
-                                .clone()
-                                .into_iter()
-                                .collect::<BTreeMap<String, String>>(),
-                        )
-                        .revision(
-                            claims_token
-                                .as_ref()
-                                .and_then(|claims| claims.claims.metadata.as_ref())
-                                .and_then(|jwt::CapabilityProvider { rev, .. }| *rev)
-                                .unwrap_or_default(),
-                        )
-                        .build()
-                        .expect("failed to build provider description")
-                },
-            )
-            .collect();
+    async fn inventory(&self, lattice: &str) -> anyhow::Result<HostInventory> {
+        trace!(lattice = lattice, "generating host inventory");
 
+        let lattice = self.lattices.get(lattice).context("lattice not found")?;
```

Comment:
Since this is the error a user would see (even though it probably shouldn't be subscribing here)
```suggestion
        let lattice = self.lattices.get(lattice).with_context(|| format!("host is not in lattice {lattice}, inventory is empty"))?;
```